evalml/tests/automl_tests/test_automl.py

```diff
@@ -3744,99 +3748,81 @@ def test_automl_respects_pipeline_parameters_with_duplicate_components(
-        "Random Forest Classifier": ["Random Forest Classifier", "One Hot Encoder_1"],
+        "One Hot Encoder": ["One Hot Encoder", "X", "y"],
+        "One Hot Encoder_1": ["One Hot Encoder", "One Hot Encoder.x", "y"],
+        "Random Forest Classifier": [
+            "Random Forest Classifier",
+            "One Hot Encoder_1.x",
+            "y",
+        ],
     }
-    component_graph_linear = [
-        "One Hot Encoder",
-        "One Hot Encoder",
-        "Random Forest Classifier",
-    ]
 
     automl = AutoMLSearch(
         X,
         y,
         problem_type="binary",
         allowed_component_graphs={
             "Pipeline from dict": component_graph_dict,
-            "Pipeline from linear": component_graph_linear,
         },
         pipeline_parameters={
             "One Hot Encoder": {"top_n": 15},
             "One Hot Encoder_1": {"top_n": 25},
         },
         optimize_thresholds=False,
-        max_batches=3,
+        max_batches=1,
     )
     with env.test_context(score_return_value={automl.objective.name: 0.63}):
         automl.search()
-    for row in automl.full_rankings.iloc[1:3].parameters:
+    for row in automl.full_rankings.iloc[1:].parameters:
         assert row["One Hot Encoder"]["top_n"] == 15
         assert row["One Hot Encoder_1"]["top_n"] == 25
 
 
-@pytest.mark.parametrize("graph_type", ["linear", "dict"])
 def test_automl_respects_pipeline_custom_hyperparameters_with_duplicate_components(
-    AutoMLTestEnv, graph_type, X_y_binary
+    AutoMLTestEnv, X_y_binary
 ):
     X, y = X_y_binary
-
-    if graph_type == "linear":
-        custom_hyperparameters = {
-            "Imputer": {"numeric_impute_strategy": Categorical(["mean"])},
-            "Imputer_1": {
-                "numeric_impute_strategy": Categorical(["most_frequent", "mean"])
-            },
-            "Random Forest Classifier": {"n_estimators": Categorical([100, 125])},
-        }
-        component_graph = {
-            "Name_linear": ["Imputer", "Imputer", "Random Forest Classifier"]
-        }
-    else:
-        custom_hyperparameters = {
-            "Imputer": {
-                "numeric_impute_strategy": Categorical(["most_frequent", "mean"])
-            },
-            "Imputer_1": {"numeric_impute_strategy": Categorical(["median", "mean"])},
-            "Random Forest Classifier": {"n_estimators": Categorical([50, 100])},
-        }
-        component_graph = {
-            "Name_dict": {
-                "Imputer": ["Imputer"],
-                "Imputer_1": ["Imputer", "Imputer"],
-                "Random Forest Classifier": ["Random Forest Classifier", "Imputer_1"],
-            }
+    custom_hyperparameters = {
+        "Imputer": {"numeric_impute_strategy": Categorical(["most_frequent", "mean"])},
+        "Imputer_1": {"numeric_impute_strategy": Categorical(["median", "mean"])},
+        "Random Forest Classifier": {"n_estimators": Categorical([50, 100])},
```

Comment:
How'd you get this past black without it exploding into 20+ lines?

Reply:
LOL I just double checked: my line is 88 chars long, and black's limit is 88 chars per line--so I just bareeeely made it 😅